GPU available: True (cuda), used: True
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
HPU available: False, using: 0 HPUs
Initializing distributed: GLOBAL_RANK: 7, MEMBER: 8/10
Initializing distributed: GLOBAL_RANK: 6, MEMBER: 7/10
Initializing distributed: GLOBAL_RANK: 3, MEMBER: 4/10
Initializing distributed: GLOBAL_RANK: 2, MEMBER: 3/10
Initializing distributed: GLOBAL_RANK: 4, MEMBER: 5/10
Initializing distributed: GLOBAL_RANK: 9, MEMBER: 10/10
Initializing distributed: GLOBAL_RANK: 5, MEMBER: 6/10
Initializing distributed: GLOBAL_RANK: 0, MEMBER: 1/10
Initializing distributed: GLOBAL_RANK: 8, MEMBER: 9/10
Initializing distributed: GLOBAL_RANK: 1, MEMBER: 2/10
----------------------------------------------------------------------------------------------------
distributed_backend=nccl
All distributed processes registered. Starting with 10 processes
----------------------------------------------------------------------------------------------------

You are using a CUDA device ('NVIDIA A100-PCIE-40GB') that has Tensor Cores. To properly utilize them, you should set `torch.set_float32_matmul_precision('medium' | 'high')` which will trade-off precision for performance. For more details, read https://pytorch.org/docs/stable/generated/torch.set_float32_matmul_precision.html#torch.set_float32_matmul_precision
/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/callbacks/model_checkpoint.py:612: UserWarning: Checkpoint directory /scratch/user/zshuying/ppi_mutation/logs/esm_finetune_ddp/esm2_t36_3B_UR50D/lr4-05_unfreeze10/checkpoints exists and is not empty.
  rank_zero_warn(f"Checkpoint directory {dirpath} exists and is not empty.")
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0,1]
LOCAL_RANK: 1 - CUDA_VISIBLE_DEVICES: [0,1]
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0,1]
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0,1]
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0,1]
LOCAL_RANK: 1 - CUDA_VISIBLE_DEVICES: [0,1]
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0,1]
LOCAL_RANK: 1 - CUDA_VISIBLE_DEVICES: [0,1]
LOCAL_RANK: 1 - CUDA_VISIBLE_DEVICES: [0,1]
LOCAL_RANK: 1 - CUDA_VISIBLE_DEVICES: [0,1]

  | Name      | Type            | Params
----------------------------------------------
0 | esm_model | ESM2            | 2.8 B 
1 | proj      | Sequential      | 7.7 K 
2 | auroc     | MulticlassAUROC | 0     
----------------------------------------------
793 M     Trainable params
2.0 B     Non-trainable params
2.8 B     Total params
11,356.053Total estimated model params size (MB)
Traceback (most recent call last):
  File "/scratch/user/zshuying/ppi_mutation/scripts/myesm/ESM_finetune_ddp.py", line 85, in <module>
Traceback (most recent call last):
  File "/scratch/user/zshuying/ppi_mutation/scripts/myesm/ESM_finetune_ddp.py", line 85, in <module>
    trainer.fit(myesm,datamodule=proData)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/trainer.py", line 520, in fit
    trainer.fit(myesm,datamodule=proData)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/trainer.py", line 520, in fit
    call._call_and_handle_interrupt(
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/call.py", line 42, in _call_and_handle_interrupt
    call._call_and_handle_interrupt(
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/call.py", line 42, in _call_and_handle_interrupt
    return trainer.strategy.launcher.launch(trainer_fn, *args, trainer=trainer, **kwargs)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/strategies/launchers/subprocess_script.py", line 92, in launch
    return trainer.strategy.launcher.launch(trainer_fn, *args, trainer=trainer, **kwargs)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/strategies/launchers/subprocess_script.py", line 92, in launch
    return function(*args, **kwargs)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/trainer.py", line 559, in _fit_impl
    return function(*args, **kwargs)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/trainer.py", line 559, in _fit_impl
    self._run(model, ckpt_path=ckpt_path)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/trainer.py", line 935, in _run
    self._run(model, ckpt_path=ckpt_path)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/trainer.py", line 935, in _run
    results = self._run_stage()
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/trainer.py", line 976, in _run_stage
    results = self._run_stage()
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/trainer.py", line 976, in _run_stage
    self._run_sanity_check()
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/trainer.py", line 1005, in _run_sanity_check
    self._run_sanity_check()
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/trainer.py", line 1005, in _run_sanity_check
    val_loop.run()
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/loops/utilities.py", line 177, in _decorator
    val_loop.run()
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/loops/utilities.py", line 177, in _decorator
    return loop_run(self, *args, **kwargs)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/loops/evaluation_loop.py", line 98, in run
    return loop_run(self, *args, **kwargs)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/loops/evaluation_loop.py", line 98, in run
    self.setup_data()
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/loops/evaluation_loop.py", line 169, in setup_data
    self.setup_data()
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/loops/evaluation_loop.py", line 169, in setup_data
    dl = _process_dataloader(trainer, dl)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/connectors/data_connector.py", line 487, in _process_dataloader
    dl = _process_dataloader(trainer, dl)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/connectors/data_connector.py", line 487, in _process_dataloader
    dataloader = trainer._data_connector._prepare_dataloader(dataloader, shuffle=is_shuffled, mode=stage)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/connectors/data_connector.py", line 199, in _prepare_dataloader
    dataloader = trainer._data_connector._prepare_dataloader(dataloader, shuffle=is_shuffled, mode=stage)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/connectors/data_connector.py", line 199, in _prepare_dataloader
    sampler = self._resolve_sampler(dataloader, shuffle=shuffle, mode=mode)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/connectors/data_connector.py", line 210, in _resolve_sampler
    sampler = self._resolve_sampler(dataloader, shuffle=shuffle, mode=mode)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/connectors/data_connector.py", line 210, in _resolve_sampler
    sampler = _get_distributed_sampler(
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/connectors/data_connector.py", line 249, in _get_distributed_sampler
    sampler = _get_distributed_sampler(
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/connectors/data_connector.py", line 249, in _get_distributed_sampler
    sampler = cls(dataloader.sampler, **kwargs)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/fabric/utilities/distributed.py", line 312, in __init__
    sampler = cls(dataloader.sampler, **kwargs)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/fabric/utilities/distributed.py", line 312, in __init__
    super().__init__(_DatasetSamplerWrapper(sampler), *args, **kwargs)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/torch/utils/data/distributed.py", line 74, in __init__
    super().__init__(_DatasetSamplerWrapper(sampler), *args, **kwargs)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/torch/utils/data/distributed.py", line 74, in __init__
    raise ValueError(
ValueError: Invalid rank 7, rank should be in the interval [0, 1]
    raise ValueError(
ValueError: Invalid rank 6, rank should be in the interval [0, 1]
Traceback (most recent call last):
  File "/scratch/user/zshuying/ppi_mutation/scripts/myesm/ESM_finetune_ddp.py", line 85, in <module>
    trainer.fit(myesm,datamodule=proData)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/trainer.py", line 520, in fit
    call._call_and_handle_interrupt(
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/call.py", line 42, in _call_and_handle_interrupt
    return trainer.strategy.launcher.launch(trainer_fn, *args, trainer=trainer, **kwargs)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/strategies/launchers/subprocess_script.py", line 92, in launch
    return function(*args, **kwargs)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/trainer.py", line 559, in _fit_impl
    self._run(model, ckpt_path=ckpt_path)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/trainer.py", line 935, in _run
    results = self._run_stage()
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/trainer.py", line 976, in _run_stage
    self._run_sanity_check()
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/trainer.py", line 1005, in _run_sanity_check
    val_loop.run()
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/loops/utilities.py", line 177, in _decorator
    return loop_run(self, *args, **kwargs)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/loops/evaluation_loop.py", line 98, in run
    self.setup_data()
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/loops/evaluation_loop.py", line 169, in setup_data
    dl = _process_dataloader(trainer, dl)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/connectors/data_connector.py", line 487, in _process_dataloader
    dataloader = trainer._data_connector._prepare_dataloader(dataloader, shuffle=is_shuffled, mode=stage)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/connectors/data_connector.py", line 199, in _prepare_dataloader
    sampler = self._resolve_sampler(dataloader, shuffle=shuffle, mode=mode)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/connectors/data_connector.py", line 210, in _resolve_sampler
    sampler = _get_distributed_sampler(
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/connectors/data_connector.py", line 249, in _get_distributed_sampler
    sampler = cls(dataloader.sampler, **kwargs)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/fabric/utilities/distributed.py", line 312, in __init__
    super().__init__(_DatasetSamplerWrapper(sampler), *args, **kwargs)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/torch/utils/data/distributed.py", line 74, in __init__
    raise ValueError(
ValueError: Invalid rank 2, rank should be in the interval [0, 1]
Traceback (most recent call last):
  File "/scratch/user/zshuying/ppi_mutation/scripts/myesm/ESM_finetune_ddp.py", line 85, in <module>
    trainer.fit(myesm,datamodule=proData)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/trainer.py", line 520, in fit
    call._call_and_handle_interrupt(
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/call.py", line 42, in _call_and_handle_interrupt
    return trainer.strategy.launcher.launch(trainer_fn, *args, trainer=trainer, **kwargs)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/strategies/launchers/subprocess_script.py", line 92, in launch
    return function(*args, **kwargs)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/trainer.py", line 559, in _fit_impl
    self._run(model, ckpt_path=ckpt_path)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/trainer.py", line 935, in _run
    results = self._run_stage()
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/trainer.py", line 976, in _run_stage
    self._run_sanity_check()
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/trainer.py", line 1005, in _run_sanity_check
    val_loop.run()
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/loops/utilities.py", line 177, in _decorator
    return loop_run(self, *args, **kwargs)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/loops/evaluation_loop.py", line 98, in run
    self.setup_data()
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/loops/evaluation_loop.py", line 169, in setup_data
    dl = _process_dataloader(trainer, dl)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/connectors/data_connector.py", line 487, in _process_dataloader
    dataloader = trainer._data_connector._prepare_dataloader(dataloader, shuffle=is_shuffled, mode=stage)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/connectors/data_connector.py", line 199, in _prepare_dataloader
    sampler = self._resolve_sampler(dataloader, shuffle=shuffle, mode=mode)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/connectors/data_connector.py", line 210, in _resolve_sampler
    sampler = _get_distributed_sampler(
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/connectors/data_connector.py", line 249, in _get_distributed_sampler
    sampler = cls(dataloader.sampler, **kwargs)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/fabric/utilities/distributed.py", line 312, in __init__
    super().__init__(_DatasetSamplerWrapper(sampler), *args, **kwargs)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/torch/utils/data/distributed.py", line 74, in __init__
    raise ValueError(
ValueError: Invalid rank 4, rank should be in the interval [0, 1]
Traceback (most recent call last):
  File "/scratch/user/zshuying/ppi_mutation/scripts/myesm/ESM_finetune_ddp.py", line 85, in <module>
    trainer.fit(myesm,datamodule=proData)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/trainer.py", line 520, in fit
    call._call_and_handle_interrupt(
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/call.py", line 42, in _call_and_handle_interrupt
    return trainer.strategy.launcher.launch(trainer_fn, *args, trainer=trainer, **kwargs)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/strategies/launchers/subprocess_script.py", line 92, in launch
    return function(*args, **kwargs)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/trainer.py", line 559, in _fit_impl
    self._run(model, ckpt_path=ckpt_path)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/trainer.py", line 935, in _run
    results = self._run_stage()
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/trainer.py", line 976, in _run_stage
    self._run_sanity_check()
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/trainer.py", line 1005, in _run_sanity_check
    val_loop.run()
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/loops/utilities.py", line 177, in _decorator
    return loop_run(self, *args, **kwargs)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/loops/evaluation_loop.py", line 98, in run
    self.setup_data()
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/loops/evaluation_loop.py", line 169, in setup_data
    dl = _process_dataloader(trainer, dl)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/connectors/data_connector.py", line 487, in _process_dataloader
    dataloader = trainer._data_connector._prepare_dataloader(dataloader, shuffle=is_shuffled, mode=stage)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/connectors/data_connector.py", line 199, in _prepare_dataloader
    sampler = self._resolve_sampler(dataloader, shuffle=shuffle, mode=mode)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/connectors/data_connector.py", line 210, in _resolve_sampler
    sampler = _get_distributed_sampler(
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/connectors/data_connector.py", line 249, in _get_distributed_sampler
    sampler = cls(dataloader.sampler, **kwargs)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/fabric/utilities/distributed.py", line 312, in __init__
    super().__init__(_DatasetSamplerWrapper(sampler), *args, **kwargs)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/torch/utils/data/distributed.py", line 74, in __init__
    raise ValueError(
ValueError: Invalid rank 5, rank should be in the interval [0, 1]
Traceback (most recent call last):
  File "/scratch/user/zshuying/ppi_mutation/scripts/myesm/ESM_finetune_ddp.py", line 85, in <module>
    trainer.fit(myesm,datamodule=proData)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/trainer.py", line 520, in fit
    call._call_and_handle_interrupt(
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/call.py", line 42, in _call_and_handle_interrupt
    return trainer.strategy.launcher.launch(trainer_fn, *args, trainer=trainer, **kwargs)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/strategies/launchers/subprocess_script.py", line 92, in launch
    return function(*args, **kwargs)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/trainer.py", line 559, in _fit_impl
    self._run(model, ckpt_path=ckpt_path)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/trainer.py", line 935, in _run
    results = self._run_stage()
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/trainer.py", line 976, in _run_stage
    self._run_sanity_check()
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/trainer.py", line 1005, in _run_sanity_check
    val_loop.run()
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/loops/utilities.py", line 177, in _decorator
    return loop_run(self, *args, **kwargs)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/loops/evaluation_loop.py", line 98, in run
    self.setup_data()
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/loops/evaluation_loop.py", line 169, in setup_data
Traceback (most recent call last):
  File "/scratch/user/zshuying/ppi_mutation/scripts/myesm/ESM_finetune_ddp.py", line 85, in <module>
    dl = _process_dataloader(trainer, dl)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/connectors/data_connector.py", line 487, in _process_dataloader
    trainer.fit(myesm,datamodule=proData)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/trainer.py", line 520, in fit
    dataloader = trainer._data_connector._prepare_dataloader(dataloader, shuffle=is_shuffled, mode=stage)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/connectors/data_connector.py", line 199, in _prepare_dataloader
    call._call_and_handle_interrupt(
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/call.py", line 42, in _call_and_handle_interrupt
    sampler = self._resolve_sampler(dataloader, shuffle=shuffle, mode=mode)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/connectors/data_connector.py", line 210, in _resolve_sampler
    return trainer.strategy.launcher.launch(trainer_fn, *args, trainer=trainer, **kwargs)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/strategies/launchers/subprocess_script.py", line 92, in launch
    sampler = _get_distributed_sampler(
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/connectors/data_connector.py", line 249, in _get_distributed_sampler
    return function(*args, **kwargs)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/trainer.py", line 559, in _fit_impl
    sampler = cls(dataloader.sampler, **kwargs)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/fabric/utilities/distributed.py", line 312, in __init__
    self._run(model, ckpt_path=ckpt_path)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/trainer.py", line 935, in _run
    super().__init__(_DatasetSamplerWrapper(sampler), *args, **kwargs)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/torch/utils/data/distributed.py", line 74, in __init__
    raise ValueError(
ValueError: Invalid rank 9, rank should be in the interval [0, 1]
    results = self._run_stage()
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/trainer.py", line 976, in _run_stage
    self._run_sanity_check()
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/trainer.py", line 1005, in _run_sanity_check
    val_loop.run()
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/loops/utilities.py", line 177, in _decorator
    return loop_run(self, *args, **kwargs)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/loops/evaluation_loop.py", line 98, in run
    self.setup_data()
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/loops/evaluation_loop.py", line 169, in setup_data
    dl = _process_dataloader(trainer, dl)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/connectors/data_connector.py", line 487, in _process_dataloader
    dataloader = trainer._data_connector._prepare_dataloader(dataloader, shuffle=is_shuffled, mode=stage)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/connectors/data_connector.py", line 199, in _prepare_dataloader
    sampler = self._resolve_sampler(dataloader, shuffle=shuffle, mode=mode)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/connectors/data_connector.py", line 210, in _resolve_sampler
    sampler = _get_distributed_sampler(
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/connectors/data_connector.py", line 249, in _get_distributed_sampler
    sampler = cls(dataloader.sampler, **kwargs)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/fabric/utilities/distributed.py", line 312, in __init__
    super().__init__(_DatasetSamplerWrapper(sampler), *args, **kwargs)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/torch/utils/data/distributed.py", line 74, in __init__
    raise ValueError(
ValueError: Invalid rank 8, rank should be in the interval [0, 1]
Traceback (most recent call last):
  File "/scratch/user/zshuying/ppi_mutation/scripts/myesm/ESM_finetune_ddp.py", line 85, in <module>
    trainer.fit(myesm,datamodule=proData)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/trainer.py", line 520, in fit
    call._call_and_handle_interrupt(
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/call.py", line 42, in _call_and_handle_interrupt
    return trainer.strategy.launcher.launch(trainer_fn, *args, trainer=trainer, **kwargs)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/strategies/launchers/subprocess_script.py", line 92, in launch
    return function(*args, **kwargs)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/trainer.py", line 559, in _fit_impl
    self._run(model, ckpt_path=ckpt_path)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/trainer.py", line 935, in _run
    results = self._run_stage()
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/trainer.py", line 976, in _run_stage
    self._run_sanity_check()
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/trainer.py", line 1005, in _run_sanity_check
    val_loop.run()
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/loops/utilities.py", line 177, in _decorator
    return loop_run(self, *args, **kwargs)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/loops/evaluation_loop.py", line 98, in run
    self.setup_data()
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/loops/evaluation_loop.py", line 169, in setup_data
    dl = _process_dataloader(trainer, dl)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/connectors/data_connector.py", line 487, in _process_dataloader
    dataloader = trainer._data_connector._prepare_dataloader(dataloader, shuffle=is_shuffled, mode=stage)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/connectors/data_connector.py", line 199, in _prepare_dataloader
    sampler = self._resolve_sampler(dataloader, shuffle=shuffle, mode=mode)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/connectors/data_connector.py", line 210, in _resolve_sampler
    sampler = _get_distributed_sampler(
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/pytorch/trainer/connectors/data_connector.py", line 249, in _get_distributed_sampler
    sampler = cls(dataloader.sampler, **kwargs)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/lightning/fabric/utilities/distributed.py", line 312, in __init__
    super().__init__(_DatasetSamplerWrapper(sampler), *args, **kwargs)
  File "/scratch/user/zshuying/.conda/envs/yuning/lib/python3.9/site-packages/torch/utils/data/distributed.py", line 74, in __init__
    raise ValueError(
ValueError: Invalid rank 3, rank should be in the interval [0, 1]
srun: error: g024: task 7: Exited with exit code 1
srun: error: g024: task 6: Exited with exit code 1
srun: error: g026: task 9: Exited with exit code 1
srun: error: g022: task 2: Exited with exit code 1
srun: error: g023: task 4: Exited with exit code 1
srun: error: g023: task 5: Exited with exit code 1
srun: error: g022: task 3: Exited with exit code 1
srun: error: g026: task 8: Exited with exit code 1
